src/streamlink/plugins/ltv_lsm_lv.py

```diff
@@ -1,45 +1,83 @@
 import logging
 import re
-from urllib.parse import urlparse
 
 from streamlink.plugin import Plugin, pluginmatcher
-from streamlink.plugin.api.utils import itertags
+from streamlink.plugin.api import validate
 from streamlink.stream.hls import HLSStream
 
 log = logging.getLogger(__name__)
 
 
 @pluginmatcher(re.compile(
-    r"https?://ltv\.lsm\.lv/lv/tieshraide"
+    r"https://ltv\.lsm\.lv/lv/tiesraide"
 ))
 class LtvLsmLv(Plugin):
     """
     Support for Latvian live channels streams on ltv.lsm.lv
     """
+
+    API_URL = "https://player.cloudycdn.services/player/ltvlive/channel/{channel_id}/"
+    _json_data_re = re.compile(r'teliaPlayer\((\{.*?\})\);', re.DOTALL)
+
+    main_page_schema = validate.Schema(
+        validate.parse_html(),
+        validate.xml_xpath_string(".//iframe[contains(@src, 'ltv.lsm.lv/embed')][1]/@src"),
+        validate.url()
+    )
+
+    embed_code_schema = validate.Schema(
+        validate.parse_html(),
+        validate.xml_xpath_string(".//live/@*"),
+        validate.parse_json(),
+        {"source": {"embed_code": validate.text}},
+        validate.get("source"), validate.get("embed_code"),
+        validate.parse_html(),
+        validate.xml_xpath_string(".//iframe/@src"),
```

Comment:
XPath queries always return lists of nodes. The `xml_xpath_string` wrapper simply turns the query into `str(query)`, but that doesn't change the fact that a query can return multiple nodes. Without limiting the number of nodes to one via the `[1]` predicate, the query could return nonsense data.

```py
validate.xml_xpath_string(".//iframe[@src][1]/@src"),
```

The `.//live/@*` query from above has the same issue and needs to be fixed with a more strict query.

Reply:
The issue with the `live` tag, and why the xpath is the way it is, is because the attribute name is `:embed-data` which I could not get to work any other way.

Reply:
Does `@*[name()=':embed-data']` work? This selects all attributes and then filters them by name.

Reply:
On which page can I find the `live` tag on their site?

Reply:
yeah, `@*[name()=':embed-data']` works, nice.